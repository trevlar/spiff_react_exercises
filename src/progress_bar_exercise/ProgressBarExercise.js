import React from "react";
import Exercise from "../exercise/Exercise";
import ProgressBarSolution from "./ProgressBarSolution/ProgressBarSolution";

const ProgressBarExercise = () => {
  return (
    <div className="progress-bar-exercise">
      <Exercise
        solution={<ProgressBarSolution />}
        specsUrl="https://github.com/SpiffInc/spiff_react_exercises/issues/1"
        title="Progress Bar Exercise"
      />
    </div>
  );
};

export default ProgressBarExercise;
