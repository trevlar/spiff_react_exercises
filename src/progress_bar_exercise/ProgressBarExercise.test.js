import React from 'react';
import {render} from '@testing-library/react'
import ProgressBarExercise from './ProgressBarExercise';

describe('ProgressBar', () => {
    it('renders properly with expected snapshot', () => {
        const progressBarExercise = render((
            <ProgressBarExercise solution={<div />} specsUrl="test.com" title="Progress Bar Exercise Test"/>
        ));
        expect(progressBarExercise).toMatchSnapshot()
    })
})
