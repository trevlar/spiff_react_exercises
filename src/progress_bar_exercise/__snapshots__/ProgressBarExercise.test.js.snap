// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar renders properly with expected snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="progress-bar-exercise"
      >
        <div
          class="exercise"
        >
          <h2>
            Progress Bar Exercise
          </h2>
          <div
            class="specs-url block"
          >
            <h4>
              Specs
            </h4>
            <a
              href="https://github.com/SpiffInc/spiff_react_exercises/issues/1"
            >
              https://github.com/SpiffInc/spiff_react_exercises/issues/1
            </a>
          </div>
          <div
            class="solution block"
          >
            <h4>
              Solution
            </h4>
            <div
              class="solution-workspace"
            >
              <div
                class="progress-container"
              >
                <div
                  class="progress-bar"
                  style="width: 400px;"
                >
                  <div
                    class="progress"
                    style="width: 0px;"
                  />
                </div>
                 
                <span>
                   
                  0
                  %
                </span>
              </div>
              <button
                class="btn btn-success"
              >
                Start Request
              </button>
              <button
                class="btn btn-warning"
              >
                Finish Request
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="progress-bar-exercise"
    >
      <div
        class="exercise"
      >
        <h2>
          Progress Bar Exercise
        </h2>
        <div
          class="specs-url block"
        >
          <h4>
            Specs
          </h4>
          <a
            href="https://github.com/SpiffInc/spiff_react_exercises/issues/1"
          >
            https://github.com/SpiffInc/spiff_react_exercises/issues/1
          </a>
        </div>
        <div
          class="solution block"
        >
          <h4>
            Solution
          </h4>
          <div
            class="solution-workspace"
          >
            <div
              class="progress-container"
            >
              <div
                class="progress-bar"
                style="width: 400px;"
              >
                <div
                  class="progress"
                  style="width: 0px;"
                />
              </div>
               
              <span>
                 
                0
                %
              </span>
            </div>
            <button
              class="btn btn-success"
            >
              Start Request
            </button>
            <button
              class="btn btn-warning"
            >
              Finish Request
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
